src/Tutor.Core/BuildingBlocks/Generics/ICrudRepository.cs

```diff
@@ -9,5 +10,6 @@ public interface ICrudRepository<TEntity> where TEntity : Entity
     TEntity Create(TEntity entity);
     void BulkCreate(List<TEntity> entities);
     TEntity Update(TEntity entity);
-    void Delete(int id);
+    TEntity Update(TEntity storedEntity, TEntity entity);
+    Result Delete(int id);
```

Comment:
Should everything from the repository return Result or just void methods or just the Delete method?